README.md

```diff
@@ -1559,28 +1256,357 @@ let main ~config_changed =
+`~weight` must be `>= 0.0` and `<= 1.0`;
+Example: given an IO-bound job that averages 2% of one CPU core, pass `~weight:0.02`.
+
+Each domain worker starts new jobs until the total `~weight` of its running jobs reaches `1.0`.
+
+### Domain Manager
+
+Let's repeat the same example using the domain manager manually.
+Each call to `Domain_manager.run` creates a new domain.
+Each domain is created on the fly and discarded immediately after completion.
+
+```ocaml
+let main ~domain_mgr =
+  let test n =
+    traceln "sum 1..%d = %d" n
+      (Eio.Domain_manager.run domain_mgr
+        (fun () -> sum_to n))
+  in
+  Fiber.both
+    (fun () -> test 100000)
+    (fun () -> test 50000)
+```
+
+The output is identical:
+<!-- $MDX non-deterministic=output -->
+```ocaml
+# Eio_main.run @@ fun env ->
+  main ~domain_mgr:(Eio.Stdenv.domain_mgr env);;
++Starting CPU-intensive task...
++Starting CPU-intensive task...
++Finished
++sum 1..50000 = 1250025000
++Finished
++sum 1..100000 = 5000050000
+- : unit = ()
+```
+Note: the exact `traceln` output of this example is non-deterministic, because the OS is free to schedule domains as it likes.
+
+`run` waits for the domain to finish, but it allows other fibers to run while waiting. This is why we use `Fiber.both` to create multiple fibers.
+
+### Multicore Performance Guide
+
+Whether you use the Executor Pool or not, the following recommendations will help you extract as much performance as possible from your hardware.
+- There's a certain overhead associated with placing execution onto another domain,
+  but that overhead will be paid off quickly if your job takes at least a few milliseconds to complete.
+  Jobs that complete under 2-5ms may not be worth running on a separate domain.
+- Similarly, jobs that are 100% I/O-bound may not be worth running on a separate domain.
+  The small initial overhead is simply never recouped.
+- If your program never hits 100% CPU usage, it's unlikely that parallelizing it will improve performance.
+- Try to avoid reading or writing to memory that's modified by other domains after the start of your job.
+  Ideally, your jobs shouldn't need to interact with other domains' "working data".
+  Aim to make your jobs as independent as possible.
+  - If unavoidable, the [Saturn](https://github.com/ocaml-multicore/saturn) library offers a collection of efficient threadsafe data structures.
+  - The [Multicore Guide](./doc/multicore.md) also explains how to safely and efficiently share memory between domains
+- It's often easier to design code to be multithreading friendly from the start
+  (by making longer, independent jobs) than by refactoring existing code.
+- There's a cost associated with creating a domain, so try to use the same domains for longer periods of time. The Executor Pool takes care of this automatically.
+- Having a large number of domains active at the same time imposes additional overhead on both the OS scheduler and the OCaml runtime, even if those domains are idle.
```

Comment:
I think some of the issues with idle domains and performance were fixed in 5.1 https://github.com/ocaml/ocaml/pull/11903. It's ok to leave the note as is. If there are more issues, it might be worth reporting upstream.

Reply:
Ah, good to know!